torch/_dynamo/variables/tensor.py

```diff
@@ -849,137 +849,93 @@ class TensorWithTFOverrideVariable(VariableTracker):
     def create(
         tx,
         tensor_variable,
-        orig_tensor_variable_source,
         torch_function_fn,
         subclass_type,
         **kwargs,
     ):
         var = TensorWithTFOverrideVariable(
             tensor_variable,
-            orig_tensor_variable_source,
             torch_function_fn,
             subclass_type,
             **kwargs,
         )
         # stash the subclass type to rewrap an output tensor if needed
-        tx.output.install_global(var.global_class_name(), subclass_type)
+        if var.global_class_name() not in tx.output.global_scope:
+            tx.output.install_global(var.global_class_name(), subclass_type)
```

Comment:
This doesn't seem right.  Just because a name exists in global scope doesn't mean it point to the same the same subclass.

Reply:
I'll add the id of the class type to the name